server/src/test/java/io/crate/auth/UserAuthenticationMethodTest.java

```diff
@@ -145,25 +148,90 @@ public void test_jwt_authentication() throws Exception {
         assertThat(authenticatedRole.name()).isEqualTo(JWT_USER.name());
     }
 
+    @Test
+    public void test_jwt_authentication_default_aud_same_as_token() throws Exception {
+        // JWT_TOKEN has aud = "test_cluster_id", imitate that cluster id is same.
+        String clusterId = "test_cluster_id";
+        Roles roles = () -> List.of(
+            userOf(
+                "John",
+                Set.of(),
+                new HashSet<>(),
+                getSecureHash("johns-pwd"),
+                // User doesn't have "aud" JWT property, cluster id will be used as aud.
+                new JwtProperties("https://console.cratedb-dev.cloud/api/v2/meta/jwk/", "cloud_user", null)
+            )
+        );
+        JWTAuthenticationMethod jwtAuth = new JWTAuthenticationMethod(
+            roles,
+            jwkProviderFunction(null),
+            () -> clusterId
+        );
+        assertThat(jwtAuth.name()).isEqualTo("jwt");
+
+        Credentials credentials = new Credentials(JWT_TOKEN);
+        assertThat(credentials.username()).isNull();
+        credentials.setUsername(JWT_USER.name());
+
+        Role authenticatedRole = jwtAuth.authenticate(credentials, null);
+        assertThat(authenticatedRole).isNotNull();
```

Comment:
The following assertions will fail if `authenticationRole` is null. No strong opinion, but I'd prefer to only assert for the minimum required for the test.
```suggestion
```